designer/server/src/common/helpers/auth/azure-oidc.js

```diff
@@ -48,12 +68,22 @@ export const azureOidc = {
               'offline_access',
               'user.read'
             ],
-            profile(credentials) {
+            profile(credentials, params) {
               const artifacts = token.decode(credentials.token)
+              const idToken = params.id_token
 
               token.verifyTime(artifacts)
               token.verifyPayload(artifacts)
 
+              const idTokenPayload = /** @type {{payload: idTokenPayload}} */ (
+                token.decode(idToken).decoded
+              ).payload
+              const assignedScopes = getScopesForUserProfile(
+                idTokenPayload.groups ?? []
+              )
+
+              credentials.scope = assignedScopes
```

Comment:
I don't think so. That function is per request, rather than when the token is received. We only need to grab their groups when the user is signing in or is refreshed. My understanding of the scope handler is where you want to elevate the user's access based on a specific request, which wouldn't be the case here.

`credentials.scope` is something I'm still trying to work out. It's definitely the right way to do it, I'm a bit puzzled though at the type errors. The [docs](https://hapi.dev/api/?v=21.3.3#-routeoptionsauthaccessscope) indicate it's a property of credentials, as does a few guides I found online.

Reply:
Probably Bell using its own bespoke `request` types?

You'll see in https://github.com/DEFRA/forms-designer/pull/145#discussion_r1595102058 that `credentials.scope` is typed correctly there?

Be good to move the code into `createUserSession()` so the compiler can check our homework

```patch
      // Create user session
      const credentials = await createUserSession(request)

      if (!credentials?.user) {
        return Boom.unauthorized()
      }
      
+     if (!credentials?.scope) {
+       // Do the flashy auth failed thing
+     }
```